### PR TITLE
Change exception deep level on sendWaitingForSendingMessage

### PR DESCRIPTION
**Pull request by hagneva1**
Change exception deep level on sendWaitingForSendingMessage to try to send all messages even if one of them return an InvalidParameterException

### CODE REVIEW

**Comment by prigaux** [comment]
I'm not a big fan of catching "Exception". Maybe a little less broad exception? maybe org.esupportail.smsuapi.exceptions.InvalidParameterException ?

Also note src/main/java/org/esupportail/smsu/services/scheduler/job/SuperviseSmsSending.java catches some exceptions. That code would be useless after your change.

**Comment by hagneva1** [comment]
Add new commit with your suggestion. Also add commit for manage new state : already sent. In case, some messages get stock in Waiting for sending state after a proxypass timeout.

File: src/main/java/org/esupportail/smsu/business/SendSmsManager.java [hunk 1]
```diff
@@ -204,34 +202,39 @@ public void sendWaitingForSendingMessage() throws HttpException, InsufficientQuo
...
 
-			daoService.updateMessage(message);
+				daoService.updateMessage(message);
 
-			if (logger.isDebugEnabled()) {
-				logger.debug("End of managment of message with id : " + message.getId());
+				if (logger.isDebugEnabled()) {
+					logger.debug("End of managment of message with id : " + message.getId());
+				}
+			} catch (org.esupportail.smsuapi.exceptions.InvalidParameterException e) {
```
Comment (vbonamy) [inline src/main/java/org/esupportail/smsu/business/SendSmsManager.java]: Je pense qu'il faudrait ici avoir une exception explicite de type SmsAlreadySent qui serait renvoyer par esup-smsu-api

 -> https://github.com/EsupPortail/esup-smsu-api/blob/1c08244da1afc82e0a0e148fa6b32c9919adca9b/src/main/java/org/esupportail/smsuapi/business/SendSmsManager.java#L131

-> modifications à effectuer au niveau de esup-smsu-api pour le mapping correct de l'exception ainsi ajouté :
* https://github.com/EsupPortail/esup-smsu-api/blob/1c08244da1afc82e0a0e148fa6b32c9919adca9b/src/main/java/org/esupportail/smsuapi/services/servlet/RestServlet.java#L65
* https://github.com/EsupPortail/esup-smsu-api/blob/1c08244da1afc82e0a0e148fa6b32c9919adca9b/src/main/java/org/esupportail/smsuapi/services/client/HttpRequestSmsuapiWS.java#L168

Comment (vbonamy) [reply src/main/java/org/esupportail/smsu/business/SendSmsManager.java]: Ce qui va dans le sens d'ailleurs de l'issue suivante : https://github.com/EsupPortail/esup-smsu-api/issues/11

**Comment by hagneva1** [comment]
Change catch exception and add back catch HttpException on SuperviseSmsSending

**Pull request merged**